zephyr.py

```diff
@@ -40,8 +45,11 @@ def _fixTuple(self, item):
             item[2] = item[2][1:]
 
         if '@' not in item[2]:
-            item[2] += '@%s' % _z.realm()
+            item[2] += '@%s' % realm()
 
+        if is_py3:
+            item = [bytes(i, "utf-8") for i in item]
```

Comment:
This certainly shouldn’t be conditionally encoded; that makes the API different for different Python versions. The Python 3 version is wrong here, because you’re encoding not just the string passed to libzephyr, but also the string added to the Python `set`. One would expect that it would at least be possible (i.e. for a tuple that’s already in canonical format) for `item in subscriptions` to hold after `subscriptions.add(item)`.